src/include/main/client_context.h

```diff
@@ -1,38 +1,55 @@
 #pragma once
 
 #include <atomic>
+#include <chrono>
 #include <cstdint>
+#include <memory>
 
 #include "common/api.h"
+#include "main/kuzu_fwd.h"
 
 namespace kuzu {
 namespace main {
 
+struct ActiveQuery {
+    explicit ActiveQuery();
+
+    std::chrono::time_point<std::chrono::high_resolution_clock> startTime;
+    std::atomic<bool> interrupted;
+};
+
 /**
  * @brief Contain client side configuration. We make profiler associated per query, so profiler is
  * not maintained in client context.
  */
 class ClientContext {
     friend class Connection;
+    friend class testing::TinySnbDDLTest;
+    friend class testing::TinySnbCopyCSVTransactionTest;
 
 public:
-    /**
-     * @brief Constructs the ClientContext object.
-     */
-    KUZU_API explicit ClientContext();
-    /**
-     * @brief Deconstructs the ClientContext object.
-     */
-    KUZU_API ~ClientContext() = default;
-
-    /**
-     * @brief Returns whether the current query is interrupted or not.
-     */
-    KUZU_API bool isInterrupted() const { return interrupted; }
+    explicit ClientContext();
+
+    ~ClientContext() = default;
+
+    inline void interrupt() { activeQuery->interrupted = true; }
+
+    bool isInterrupted() const { return activeQuery->interrupted; }
+
+    inline bool isQueryTimeOut() {
+        auto currentTime = std::chrono::high_resolution_clock::now();
+        return std::chrono::duration_cast<std::chrono::milliseconds>(
+                   currentTime - activeQuery->startTime)
+                   .count() > timeoutInMS;
+    }
+
+    inline uint64_t getQueryTimeOut() const { return timeoutInMS; }
 
 private:
     uint64_t numThreadsForExecution;
-    std::atomic<bool> interrupted;
+    std::unique_ptr<ActiveQuery> activeQuery;
+    // The default query timeout is 120 seconds.
+    uint64_t timeoutInMS = 120000;
```

Comment:
I don't know if constantly do the timing will add overhead to us. So I wanna to have a flag checking if timeout is  set/enabled. In this case, if timeout is not enabled, the branch prediction will always succeed and we don't pay any cost.

Reply:
added an option to enable/disable timeout